modules/random-list/random-list.nu

```diff
@@ -0,0 +1,115 @@
+# Generate a random boolean list.
+export def "random-list bool" [
+    list_length: int, # A length of the list
+    bias: float = 0.5 # A probability of "true"
+] {
+    if $bias < 0 or $bias > 1 {
+        error make {
+            msg: "invalid probability: must be between 0 and 1"
+        }
```

Comment:
applies to all the `error make` calls below :wink: 

currently, the error looks like
```
> random-list bool 10 -1
Error: 
  × invalid probability: must be between 0 and 1
   ╭─[modules/random-list/random-list.nu:6:
1]
 6 │     if $bias < 0 or $bias > 1 {
 7 │         error make {
   ·         ─────┬────
   ·              ╰── originates from here
 8 │             msg: "invalid probability: must be between 0 and 1"
   ╰────
```

---

if you define
```bash
def throw-error [
    error: string
    msg: string
    span: record<start: int, end: int>
] {
    error make {
        msg: $"(ansi red_bold)($error)(ansi reset)"
        label: {
            text: $msg
            start: $span.start
            end: $span.end
        }
    }
}
```
you can write
```
    if $bias < 0 or $bias > 1 {
        throw-error "value_error" "must be between 0 and 1" (metadata $bias | get span)
    }
```
and have the following kind of error (note that `value_error` would be in bold red!):
```
> random-list bool 10 -1
Error: 
  × value_error
   ╭─[entry #4:1:1]
 1 │ random-list bool 10 -1
   ·                     ─┬
   ·                      ╰── must be between 0 and 1
   ╰────
```

Reply:
Lol, what is it?

![image](https://github.com/nushell/nu_scripts/assets/42812113/012b14fa-51be-4205-bc80-b8a1e90f9ee9)

Why error?

Reply:
I'm guessing this is because you're on a older version i.e. not the latest main. This record syntax could've been added recently.

Reply:
I've used just `record` for now. Now errors look right. 😄

Reply:
should `nu_scripts` follow the latest `main` of `nushell` and thus use the new `record<...>` syntax?

Reply:
In the latest main either `record` or `record<...>` is valid, so I think it's fine to have either.

Reply:
I'll update script later, when update nushell. ;)